vllm/attention/layer.py

```diff
@@ -117,7 +118,10 @@ def __init__(
         compilation_config = get_current_vllm_config().compilation_config
         if prefix in compilation_config.static_forward_context:
             raise ValueError(f"Duplicate layer name: {prefix}")
-        compilation_config.static_forward_context[prefix] = self
+        # use a placeholder kv cache tensor during init, which will be replaced
+        # after kv cache initialization
+        compilation_config.static_forward_context[
+            prefix] = LayerForwardContext(self, torch.tensor([]))
```

Comment:
can we directly add an attribute to the attention layer? e.g. `self.kv_cache = torch.tensor([])` or `self.kv_cache = None` , and later set `self.kv_cache` to the correct tensor.

Reply:
Great idea. I've changed the code based on your suggestion.